arwen/elrondapi/elrondei.go

```diff
@@ -466,11 +501,21 @@ func writeLog(context unsafe.Pointer, pointer int32, length int32, topicPtr int3
 	instCtx := wasmer.IntoInstanceContext(context)
 	hostContext := arwen.GetErdContext(instCtx.Data())
 
-	log := arwen.LoadBytes(instCtx.Memory(), pointer, length)
+	log, err := arwen.LoadBytes(instCtx.Memory(), pointer, length)
+	if withFault(err, context) {
+		return
+	}
+
+	topics, err := arwen.GuardedMakeByteSlice2D(numTopics)
+	if withFault(err, context) {
+		return
+	}
 
-	topics := make([][]byte, numTopics)
 	for i := int32(0); i < numTopics; i++ {
-		topics[i] = arwen.LoadBytes(instCtx.Memory(), topicPtr+i*arwen.HashLen, arwen.HashLen)
+		topics[i], err = arwen.LoadBytes(instCtx.Memory(), topicPtr+i*arwen.HashLen, arwen.HashLen)
```

Comment:
if we change LoadBytes and StoreBytes to get the context, than fault handling can be done inside.

Reply:
It is somehow possible, but we need a similar check in the API function anyway, in order to end execution on fault, with `return`:

```
if ... {
   return ... (end API function execution)
}
```